File: 014_longest_common_prefix/main.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string result = "";
        if(strs.size()  == 0) {
            return result;
        }

        string prefix = strs[0];
        int i = 0;
        while(i < prefix.length()) {
            bool common_flg = true;
            for(int j = 0; j < strs.size(); j++) {
                if(strs[j][i] != prefix[i]) {
                    common_flg = false;
                    break;
                }
            }
            if(!common_flg) {
                break;
            }
            result += prefix[i];
            i++;
        }
        return result;
    }
};

File: 014_longest_common_prefix/main.rb
# frozen_string_literal: true

# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  return '' if strs.empty?

  prefix = strs[0]
  result = ''
  prefix.chars.each_with_index do |char, i|
    return result unless strs.all? { |str| str[i] == char }

    result += char
  end
  result
end
